ui-v2/app/styles/components/app-view/layout.scss

```diff
@@ -10,6 +10,15 @@
   display: flex;
   align-items: flex-start;
 }
+%app-view header dl {
+  float: left;
```

Comment:
Floats instead of flexbox?

Reply:
Generally I'll reach for `float` first as it will automatically 'JustWork' when you squeeze up your screen, so `float`ed elements will drop underneath each other into a column view. `flex` won't do that without you adding additional `@media` rules. Actually I can't remember whether I've looked at this properly from a responsive angle, will give it a double check.

Reply:
It should, and you can get finicky with things like `flex-grow`, `flex-shrink`, and `flex-basis`. IMO, it's better than having to deal with the ol' `clearfix`.

Reply:
Hi @DingoEatingFuzz ,

The property to use to implement this with `flex` would be `flex-wrap`. Here's what things look like when I change the CSS by switching out the `float` for a `flex`:

<img width="946" alt="screenshot 2019-02-18 at 10 00 03" src="https://user-images.githubusercontent.com/554604/53001652-13ab3d80-3423-11e9-85be-834fda1e2f09.png">

Before I go further into this. I'd be interested to know what the basis of:

> IMO, it's better than having to deal with the ol' `clearfix`.

..is?

I have my opinions on using a`<div class="clearfix">`, and just to note, I don't like using them at all, but I'm interested to know what _your reasons_ are? Could you explain _exactly why_ you don't like using "the ol' clearfix".

Thanks,

John

Reply:
The name says it all: clear**fix**. It's a workaround caused by float never being meant for layouts. Sure, it works like a charm, but `flexbox` and `grid` were introduced as proper solutions to the hole float/clearfix filled.

> The property to use to implement this with `flex` would be `flex-wrap`.

There isn't one property to use. It's going to take a combination of things. `flex-wrap` will solve the browser squeeze issue you mentioned, but it alone isn't going to get you the layout you want, as your screenshot shows.

> Before I go further into this.

It isn't a critical issue, so there is not need to go further into it unless you have time on your hands. I only bring it up because I see flexbox as the successor to this float/clearfix pattern.

Reply:
> The name says it all: clear**fix**

Your exact reasons for not using something is just because of its name?

My exact reasons for not using an extra clearfix div is because I try to avoid adding markup to templates just for styling concerns, it makes it harder to build a responsive webapp.

It's important to note here, I don't use a clear**fix** anywhere here. This is another [Straw man](https://en.wikipedia.org/wiki/Straw_man)

Completely agree it's not critical and this isn't important in the scheme of things, and I definitely don't have time on my hands! But it is related to something that keeps coming up.

I do use `clear`, a CSS Level 1 property. Using `float` and `clear` here prevent me from adding extra markup for styling concerns or over-complicating things. Those are my reasons. Under slightly different circumstances I might have considered a `flex` approach.